File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Polylines.java
package org.maplibre.android.maps;


import androidx.annotation.NonNull;

import org.maplibre.android.annotations.Polyline;
import org.maplibre.android.annotations.PolylineOptions;

import java.util.List;

/**
 * Interface that defines convenient methods for working with a {@link Polyline}'s collection.
 */
interface Polylines {
  Polyline addBy(@NonNull PolylineOptions polylineOptions, @NonNull MapLibreMap maplibreMap);

  List<Polyline> addBy(@NonNull List<PolylineOptions> polylineOptionsList, @NonNull MapLibreMap maplibreMap);

  void update(Polyline polyline);

  List<Polyline> obtainAll();
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Polylines.kt
package org.maplibre.android.maps

import org.maplibre.android.annotations.Polyline
import org.maplibre.android.annotations.PolylineOptions

/**
 * Interface that defines convenient methods for working with a [Polyline]'s collection.
 */
internal interface Polylines {
    fun addBy(
        polylineOptions: PolylineOptions,
        maplibreMap: MapLibreMap,
    ): Polyline

    fun addBy(
        polylineOptionsList: List<PolylineOptions>,
        maplibreMap: MapLibreMap,
    ): List<Polyline>

    fun update(polyline: Polyline)

    fun obtainAll(): List<Polyline>
}
